fiatconnect-api.md

```diff
@@ -383,105 +399,201 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+MUST honor login requests for both, as described below.
+
+##### 3.2.1.5.1. Parameters
+
+###### 3.2.1.5.1.1. Request Body
+
+The *plaintext* request body for this endpoint MUST correspond to a signed SIWE message (as described above) in the [EIP-191](https://eips.ethereum.org/EIPS/eip-191)
+Signed Data Standard format, as required by the SIWE specification.
+
+##### 3.2.1.5.2. Responses
+
+###### 3.2.1.5.2.1. HTTP `200`
+
+On success, the server MUST return an HTTP `200`, update the state of the user's session to be authenticated, and return a session cookie.
+
+###### 3.2.1.5.2.2. HTTP `401`
+
+On failure, the server MUST return an HTTP `400`, with the following response body.
+
+```
 {
-	"alg": "ES256",
-	"typ": "JWT"
+	error: `ErrorEnum`,
 }
 ```
 
-Servers MAY support other algorithms, but this is not required.
+##### 3.2.1.5.3. Semantics
+
+This endpoint performs validation on a signed SIWE-compliant message. In addition to checking that the signature itself is valid and corresponds to the included
+address, it also performs various safety and security checks on the contents of the message. If all checks pass, the server updates the client's session to be authenticated,
+and returns a `200`.
 
-#### 3.2.1.2. JWT Payload
+###### 3.2.1.5.3.1. Success
 
-Servers MUST recognize the following *registered claims* within the JWT payload.
+On success, the server MUST respond with a `200` status code. The server MUST only respond successfully after performing the following checks.
 
-* `exp`, or *expiration time*
-* `sub`, or *subject*
-* `iss`, or *issuer*
+* A server MUST verify that the signature of the request body is valid and corresponds to the `address` field included in the signed message. If the signature does not
+  correspond to the address field, the server MUST assume that the value given for `address` corresponds to a contract-owned account, and attempt to verify that the
+  signature is valid according to the [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) standard. If neither of these checks pass, the server MUST reject the request.
 
-Servers MUST also recognize the following *private claims* within the JWT payload.
+* A server MUST verify that the `nonce` field does not correspond to a nonce that has already been used to authenticate an ongoing, valid session.
 
-* `client`, or *client API key*
+* A server MUST verify that the `domain` and `uri` fields correspond exactly to the values expected for the server's hostname/origin.
 
-The semantics of these claims are below:
+* A server MUST verify that the `Version` and `Chain ID` lines in the signed message are exactly `1` and `42220`, respectively.
 
-##### 3.2.1.2.1. `"sub"` Claim
+* If the `issued-at` field in the signed message is before the server's current timestamp, or if the `issued-at` field is after the `expiration-time` field, the server
```

Comment:
> If the `issued-at` field in the signed message is before the server's current timestamp

If the server and client exactly agree on the current time, then wouldn't `issued-at` always be before the server's current timestamp?

Reply:
I'm confused by this

Reply:
did you mean _after_ here?

Reply:
like if the `issued-at` field is *after* the server's current timestamp, the request should fail?

Reply:
if so, I am newly concerned about clock drift and wonder if I was wrong about generating these messages purely on the client

Reply:
Yes, I believe I meant _after_ here. Clock synchronization does seem extremely important here, though I think it's out of scope for the auth update. We should probably include some separate method by which clients and servers sync on clock in a separate PR. Probably some endpoint that returns the server's clock... although, the server's clock is typically returned in the `date` header, anyways. Regardless, the exact solution for clock synchronization should be a separate ticket IMO.